subprojects/docs/src/snippets/dependencyManagement/customizingResolution-ivyMetadataRule/kotlin/build.gradle.kts

```diff
@@ -9,8 +9,18 @@ repositories {
 }
 
 // tag::ivy-component-metadata-rule[]
-abstract class IvyVariantDerivationRule : ComponentMetadataRule {
-    @Inject abstract fun getObjects(): ObjectFactory
+abstract class IvyVariantDerivationRule @Inject internal constructor(objectFactory: ObjectFactory) : ComponentMetadataRule {
+    private val jarLibraryElements: LibraryElements
+    private val libraryCategory: Category
+    private val javaRuntimeUsage: Usage
+    private val javaApiUsage: Usage
+
+    init {
+        jarLibraryElements = objectFactory.named(LibraryElements::class.java, LibraryElements.JAR)
+        libraryCategory = objectFactory.named(Category::class.java, Category.LIBRARY)
+        javaRuntimeUsage = objectFactory.named(Usage::class.java, Usage.JAVA_RUNTIME)
+        javaApiUsage = objectFactory.named(Usage::class.java, Usage.JAVA_API)
```

Comment:
I _think_ the RHS can be simplified by omitting the first argument (via a kotlin type inference + an extension method we generate):

Can you try with:
```
jarLibraryElements = objectFactory.named(LibraryElements.JAR)
libraryCategory = objectFactory.named(Category.LIBRARY)
javaRuntimeUsage = objectFactory.named(Usage.JAVA_RUNTIME)
javaApiUsage = objectFactory.named(Usage.JAVA_API)
```

Reply:
done :) https://gradle.com/s/yidbjhtru5oy6